src/segment-spinner/Segment.ts

```diff
@@ -0,0 +1,253 @@
+import {
+  useCompositeItem,
+  CompositeItemOptions,
+  CompositeItemHTMLProps,
+} from "reakit";
+import { MouseEvent, useState } from "react";
+import { mergeProps } from "@react-aria/utils";
+import { DOMProps } from "@react-types/shared";
+import { unstable_useId as useId } from "reakit";
+import { callAllHandlers } from "@chakra-ui/utils";
+import { useDateFormatter } from "@react-aria/i18n";
+import { createComponent, createHook } from "reakit-system";
+
+import { SEGMENT_KEYS } from "./__keys";
+import { isNumeric, parseNumber } from "./__utils";
+import { useSpinButton } from "../utils/useSpinButton";
+import { IDateSegment, SegmentStateReturn } from "./SegmentState";
+
+export type SegmentOptions = CompositeItemOptions &
+  Pick<
+    SegmentStateReturn,
+    | "value"
+    | "next"
+    | "setSegment"
+    | "increment"
+    | "decrement"
+    | "incrementPage"
+    | "decrementPage"
+    | "dateFormatter"
+    | "confirmPlaceholder"
+  > & {
+    segment: IDateSegment;
+    isDisabled?: boolean;
+    isReadOnly?: boolean;
+    isRequired?: boolean;
+  };
+
+export type SegmentHTMLProps = CompositeItemHTMLProps & DOMProps;
```

Comment:
```suggestion
export type SegmentHTMLProps = CompositeItemHTMLProps;
```

Is DOMProps required?

Reply:
This is not required for sure.